RFC0032.md

```diff
@@ -62,14 +56,99 @@ Specifying a prioritized list of logging channels on each call to PMIx\_Log can
 * PMIX\_LOG\_GLOBAL\_SYSLOG - "global\_syslog"
 * PMIX\_LOG\_EMAIL} - "email"
 
-and so on. Marking a given channel in the list as "required" can be done by adding ":req" to the channel name - e.g., plog\_base\_order = "local\_syslog:req,global\_syslog,email". Parsing of this MCA parameter is case-insensitive, and the parser will accept amy "required" flag that starts with "req" - e.g., "reqd" and "required". Similarly, the PMIX\_LOG\_ONCE attribute can be set by default using the "plog\_base\_log\_once" MCA parameter. Note that this is specific to the PMIx reference library and is _not_ part of the standard - users are advised to check their local implementation for similar support.
+and so on. Marking a given channel in the list as "required" can be done by adding ":req" to the channel name - e.g., plog\_base\_order = "local\_syslog:req,global\_syslog,email". Parsing of this MCA parameter is case-insensitive, and the parser will accept any "required" flag that starts with "req" - e.g., "reqd" and "required". Similarly, the PMIX\_LOG\_ONCE attribute can be set by default using the "plog\_base\_log\_once" MCA parameter. Note that this is specific to the PMIx reference library and is _not_ part of the standard - users are advised to check their local implementation for similar support.
 
 Channels that are not recognized by the PMIx library will automatically be directed to the host RM for processing. This allows for RM-proprietary channel support without committing those channel names to the PMIx Standard.
 
+## Proposed Changes
+The proposed changes covered by this RFC are organized into four categories: introduction of a new server "type", API additions, new attributes, and clarification of behavior expected from prior attributes.
+
+#### Gateway Servers
+This RFC introduces the concept of a "gateway" server - i.e., a server designated at time of PMIx\_server\_init using the new PMIX\_SERVER\_GATEWAY attribute. Gateway servers act as routers for requests that cannot be serviced on backend nodes. Typical examples include logging to email or syslog on a central system management server.
+
```

Comment:
Do we need to list the type of PMIX_SERVER_GATWAY value?   Is there any rule that "in order to support certain types of logging at least one PMIx_server_init call use specify itself as a PMIX_SERVER_GATEWAY"?   Or can the pmix convenience library still support all channel types even if no PMIX_SERVER_GATEWAY's are specified?

Reply:
I can provide the actual definition - as you can guess, it is just a bool. As currently written, you would indeed require at least one "gateway" server to use the email and global_syslog channels. I suppose one could locally attempt the email channel and then look for a gateway if it failed. I don't know a way to do that with the global_syslog channel, though, as it really has to do with how they configured syslog on their cluster. Let me ponder that one a bit.